docs/sync/azure_pipelines.md

```diff
@@ -0,0 +1,316 @@
+
+## Configure `azure-pipelines.yml`
+
+Phylum analysis of dependencies can be added to existing pipelines or on it's own with this minimal configuration:
+
+```yaml
+trigger:
+  - main
+
+jobs:
+  - job: Phylum
+    pool:
+      vmImage: ubuntu-latest
+    container: phylumio/phylum-ci:latest
+    steps:
+      - script: phylum-ci
+        displayName: Analyze dependencies with Phylum
+        env:
+          PHYLUM_API_KEY: $(PHYLUM_TOKEN)
+          AZURE_TOKEN: $(AZURE_PAT)
+```
+
+This single stage pipeline configuration contains a single container job named `Phylum`, triggered to run on pushes
+or PRs targeting the `main` branch. It does not override any of the `phylum-ci` arguments, which are all either
+optional or default to secure values.
+
+Let's take a deeper dive into each part of the configuration:
+
+### Pipeline control
+
+Choose when to run the pipeline. See the [YAML schema trigger definition][yaml_trigger] documentation for more detail.
+
+```yaml
+# This is a CI trigger that will cause the
+# pipeline to run on pushes to the `main` branch
+trigger:
+  - main
+```
+
+It is recommended to also enable PR validation for the target trigger branch(es). To do so, navigate to the branch
+policies for the desired branch (`main` in this example), and configure the [Build validation policy][build_validation]
+for that branch. For more information, see the documentation on [PR triggers for Azure Repos Git][pr_triggers] hosted
+repositories or, more broadly, [events that trigger pipelines][triggers].
+
+[yaml_trigger]: https://learn.microsoft.com/azure/devops/pipelines/yaml-schema/trigger
+[build_validation]: https://learn.microsoft.com/azure/devops/repos/git/branch-policies#build-validation
+[pr_triggers]: https://learn.microsoft.com/azure/devops/pipelines/repos/azure-repos-git#pr-triggers
+[triggers]: https://learn.microsoft.com/azure/devops/pipelines/build/triggers
+
+### Job names
+
+The job name can be named differently or included in an existing stage/job.
+
+```yaml
+jobs:
+  - job: Phylum  # Name this what you like
+```
+
+### Pool selection
+
+The pool is specified at the job level here because this is a [container job][container_job] and currently only
+`windows-2019` and `ubuntu-*` images support running containers. Keeping that restriction in mind, the pool can
+be specified at the pipeline or stage level instead. See the [YAML schema pool definition][yaml_pool] documentation
+for more detail.
+
+[container_job]: https://learn.microsoft.com/azure/devops/pipelines/process/container-phases
+[yaml_pool]: https://learn.microsoft.com/azure/devops/pipelines/yaml-schema/pool
+
+```yaml
+    pool:
+      vmImage: ubuntu-latest
+```
+
+### Docker image selection
+
+The container is specified at the job level here because this is a [container job][container_job] where all steps
+in the job are meant to run with the same image. The container can also be specified as a
```

Comment:
The thinking here was to provide a minimal example while also recognizing that the example is almost certainly not going to be used verbatim in the ADO world like it could be (and is) for the GitHub world. For instance, the example `azure-pipelines.yml` file could have been even shorter:

```yaml
# This is a pipeline with a single stage and single job
# triggers are optional (this pipeline will fire for pushes to all branches)
# the pool vmImage defaults to `ubuntu-latest` when not specified
container: phylumio/phylum-ci:latest
steps:
  - script: phylum-ci
    env:
      PHYLUM_API_KEY: $(PHYLUM_TOKEN)
      AZURE_TOKEN: $(AZURE_PAT)
```

[Azure Pipeline definitions](https://learn.microsoft.com/azure/devops/pipelines/yaml-schema/pipeline) allow for omitting the `stages` and `jobs` keywords and directly specify the [`steps`](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps?view=azure-pipelines) keyword when the pipeline consists of a single stage and a single job.

However, the reality of most pipelines is that a single `azure-pipelines.yml` file is used to define the entirety of the CI environment. This stands in contrast to the GitHub Workflows model where separate files are used to define individual workflows. The result is that it is much more likely that this Phylum Azure Pipelines integration is going to be slotted into an existing `azure-pipelines.yml` file...and that file is likely to have multiple jobs, if not also multiple stages.

The [`pool` definition](https://learn.microsoft.com/azure/devops/pipelines/yaml-schema/pool) allows for specifying a pool at the pipeline, stage, or job level, but the pool specified at the lowest level of the hierarchy is used to run the job. So, instead of assuming that a pool defined at the pipeline or stage level is going to be compatible with this integration, the choice was made to be explicit about defining it at the job level.

Reply:
Sorry. I think I got "stage level" and "step level" mixed up in my brain... I thought we could define the pool and image at the step level, but I see now that is mistaken.